File: cpp/Ram.cpp
#pragma once
#include "Komponen.cpp"
#include <iostream>
#include <string>
using namespace std;
class Ram : public Komponen {
private:
    int kapasitasGB;
    string ddr;

public:
    Ram() {
    }
    Ram(int kapasitasGB, string ddr, string merk, string nama) : Komponen(merk, nama) {
        this->kapasitasGB = kapasitasGB;
        this->ddr = ddr;
    }
    void setKapasitasGB(int kapasitasGB) {

        this->kapasitasGB = kapasitasGB;
    }
    void setDdr(float ddr) {
        this->ddr = ddr;
    }
    int getKapasitasGB() {
        return this->kapasitasGB;
    }
    string getDdr() {
        return this->ddr;
    }
    ~Ram() {
    }
};

File: python/Ram.py
from Komponen import Komponen
class Ram(Komponen) :
    def __init__(self, kapasitasGB, ddr, merk, nama):
        super().__init__(merk, nama)
        
        self.__kapasitasGB = kapasitasGB
        self.__ddr = ddr
        
    def get_kapasitasGB(self):
        return self.__kapasitasGB
    def get_ddr(self):
        return self.__ddr
    
    # Setter methods
    def set_kapasitasGB(self, kapasitasGB):
        self.__kapasitasGB = kapasitasGB
    def set_ddr(self, ddr):
        self.__ddr = ddr
        